ground/src/test/java/com/google/android/ground/persistence/remote/firebase/protobuf/LoiMutationConverterTest.kt

```diff
@@ -0,0 +1,229 @@
+ */
+
+package com.google.android.ground.persistence.remote.firebase.protobuf
+
+import com.google.android.ground.model.geometry.Coordinates
+import com.google.android.ground.model.geometry.LinearRing
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.geometry.Polygon
+import com.google.android.ground.model.locationofinterest.LOI_NAME_PROPERTY
+import com.google.android.ground.model.mutation.LocationOfInterestMutation
+import com.google.android.ground.model.mutation.Mutation
+import com.google.android.ground.proto.AuditInfo.CLIENT_TIMESTAMP_FIELD_NUMBER
+import com.google.android.ground.proto.AuditInfo.DISPLAY_NAME_FIELD_NUMBER
+import com.google.android.ground.proto.AuditInfo.SERVER_TIMESTAMP_FIELD_NUMBER
+import com.google.android.ground.proto.AuditInfo.USER_ID_FIELD_NUMBER
+import com.google.android.ground.proto.Coordinates.LATITUDE_FIELD_NUMBER
+import com.google.android.ground.proto.Coordinates.LONGITUDE_FIELD_NUMBER
+import com.google.android.ground.proto.Geometry.POINT_FIELD_NUMBER
+import com.google.android.ground.proto.Geometry.POLYGON_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.CREATED_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.GEOMETRY_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.JOB_ID_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.LAST_MODIFIED_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.PROPERTIES_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.Property.STRING_VALUE_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.SOURCE_FIELD_NUMBER
+import com.google.android.ground.proto.LocationOfInterest.SUBMISSION_COUNT_FIELD_NUMBER
+import com.google.android.ground.proto.Point.COORDINATES_FIELD_NUMBER
+import com.google.android.ground.proto.Polygon.SHELL_FIELD_NUMBER
+import com.google.common.truth.Truth.assertThat
+import com.sharedtest.FakeData
+import com.sharedtest.FakeData.LOCATION_OF_INTEREST_NAME
+import java.time.Instant
+import java.util.Date
+import kotlin.test.fail
+import org.junit.Assert.assertThrows
+import org.junit.Test
+
+class LoiMutationConverterTest {
+  @Test
+  fun `toMap() retains job ID and submission count`() {
+    val mutation = newLoiMutation()
+
+    val map = mutation.createLoiMessage(TEST_USER).toFirestoreMap()
+
+    assertThat(map[JOB_ID_FIELD_NUMBER.toString()]).isEqualTo(mutation.jobId)
+    assertThat(map[SUBMISSION_COUNT_FIELD_NUMBER.toString()]).isEqualTo(10)
+  }
+
+  @Test
+  fun `toMap() retains point geometry data`() {
+    val mutation = newLoiMutation()
+
+    val map = mutation.createLoiMessage(TEST_USER).toFirestoreMap()
+
+    val geometry = map[GEOMETRY_FIELD_NUMBER.toString()]
+    if (geometry is MutableMap<*, *>) {
+      assertThat(geometry[POINT_FIELD_NUMBER.toString()])
+        .isEqualTo(
+          mapOf(
+            COORDINATES_FIELD_NUMBER.toString() to
```

Comment:
If you can define these as `vals` at test level or in a util this might be more readable, for ex:

`val COORDINATES = COORDINATES_FIELD_NUMBER.toString()`

`mapOf(COORDINATES to ... )`

Reply:
I'll do it in a follow-up